lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1905,6 +1905,18 @@ private function doMerge($entity, array &$visited, $prevManagedCopy = null, $ass
     }
 
     /**
+     * Tests if an entity is a non initialized proxy class
```

Comment:
I think it may be clearer to flip this function and rewrite it as a positive-question:

    private function isLoaded($entity){
        return !($entity instanceof Proxy) || !$entity->__isInitialized();
    }